codehearts/remembear
# Full user database support

## Description of changes

- Provides a user data interface to add, update, remove, and get by ID
- Provides a CLI interface to manage users in the database
- Provides some mocked dependencies which can be used in unit testing

## Reason for changes 

- To support user management
- To lay down the framework for future persistent data providers

## Test steps 

1. Run automated tests

        cargo test
1. Show usage information for the binary

        cargo run help
1. Add a new user to the database, you will see uid 1 and name "Leland" output

        cargo run user add Leland
1. List all users, you should see the same user from the last command

        cargo run user list
1. Update the name of the user, you should see uid 1 and name "Bob" output

        cargo run update user 1 --name=Bob
1. List all users, you should see the same user from the last command

        cargo run user list
1. Remove the user, you should see uid 1 and name "Bob" output

        cargo run user remove 1
1. List all users to confirm the database is empty

        cargo run user list
